tools/codegen/src/main.rs

```diff
@@ -287,19 +285,40 @@ fn main() -> Result<()> {
                 }
             };
 
-            eprint!("downloading {url} for checksum ... ");
+            eprint!("downloading {url} for etag and checksum ... ");
             let download_cache = &download_cache_dir.join(format!(
                 "{version}-{platform:?}-{}",
                 Path::new(&url).file_name().unwrap().to_str().unwrap()
             ));
+            let response = download(&url)?;
+            let etag = response
```

Comment:
According to [mdn doc for etags](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag), it seems that the etag could contain a `W/` to indicate weak validator is used.

I think we should strip it from the etag stored in manifest.

Example etag:

```
ETag: "33a64df551425fcc55e4d42a148795d9f25f89d4"
ETag: W/"0815"
```

Reply:
the docs for github are clear that asset etags are not weak validators.